spec.bs

```diff
@@ -716,7 +716,54 @@ To <dfn>generate and score bids</dfn> given an [=auction config=] |auctionConfig
       1. [=list/For each=] |ig| of |groups|:
         1. TODO: If rerunning `generateBid()` with only k-anonymous ads, remember to swap out |ig|'s ads
           to just k-anonymous ones so that [=validating an ad url=] excludes the non-k-anonymous ones.
-        1. TODO: Let |interestGroup| be ... from |ig| ... minus priority and prioritySignalsOverrides and any browser-defined pieces
+        1. Let |igIDL| be a new {{AuctionAdInterestGroup}}.

```

Comment:
The type you pass to `generateBid` is more precise than AuctionAdInterestGroup, so it'll help developers' IDEs if you define that narrower type.

Reply:
defined a new IDL {{GenerateBidInterestGroup}}.
Feel free to suggest a better name :)

Reply:
I also see the couple `any` fields that are JSON strings by the time they get stored in the browser or passed to bidding JS. `executionMode` could probably also be an enum in the outbound dictionary, and you might find other type refinements that would be helpful to developers. Refining the types is also extra helpful if you can use this dictionary as the internal storage type, since they mean spec readers don't have to worry about other kinds of values.

You could use [dictionary inheritance](https://webidl.spec.whatwg.org/#dfn-inherit-dictionary) to avoid repeating the fields that are actually the same.